### ISSUE TITLE
Refactor this method to reduce its Cognitive Complexity from 33 to the 15 allowed.

### ISSUE THREAD

**Issue by am-elbah** [rilling/OpenTracks-Concordia#103]
**Describe the technical debt**
15 switch cases had the same if statement condition checked inside them all.

**Type of technical debt**
Code debt (Cognitive Complexity)

**Location**
src/main/java/de/dennisguse/opentracks/io/file/importer/GpxTrackImporter.java

**To Reproduce**
have the same if statement condition used in all 15 switch cases.

1. Copy paste the code fragment which contains the technical debt.

 public void endElement(String uri, String localName, String tag) {
        switch (tag) {
            case TAG_GPX:
                onFileEnd();
                break;
            case TAG_MARKER:
                onMarkerEnd();
                break;
            case TAG_TRACK:
                trackImporter.setTrack(context, name, uuid, description, category, null, zoneOffset);
                zoneOffset = null;
                break;
            case TAG_TRACK_SEGMENT:
                onTrackSegmentEnd();
                break;
            case TAG_TRACK_POINT:
                currentSegment.add(createTrackPoint());
                break;
            case TAG_NAME:
                if (content != null) {
                    name = content.trim();
                }
                break;
            case TAG_DESCRIPTION:
                if (content != null) {
                    description = content.trim();
                }
                break;
            case TAG_TYPE:
                if (content != null) {
                    category = content.trim();
                }
                break;
            case TAG_TIME:
                if (content != null) {
                    time = content.trim();
                }
                break;
            case TAG_ALTITUDE:
                if (content != null) {
                    altitude = content.trim();
                }
                break;
            case TAG_COMMENT:
                if (content != null) {
                    markerType = content.trim();
                }
                break;
            case TAG_EXTENSION_SPEED:
            case TAG_EXTENSION_SPEED_COMPAT:
                if (content != null) {
                    speed = content.trim();
                }
                break;
            case TAG_EXTENSION_HEARTRATE:
                if (content != null) {
                    heartrate = content.trim();
                }
                break;
            case TAG_EXTENSION_CADENCE:
                if (content != null) {
                    cadence = content.trim();
                }
                break;
            case TAG_EXTENSION_POWER:
                if (content != null) {
                    power = content.trim();
                }
                break;
            case TAG_ID:
                if (content != null) {
                    uuid = content.trim();
                }
                break;
            case TAG_EXTENSION_GAIN:
                if (content != null) {
                    gain = content.trim();
                }
                break;
            case TAG_EXTENSION_LOSS:
                if (content != null) {
                    loss = content.trim();
                }
                break;
            case TAG_EXTENSION_DISTANCE:
                if (content != null) {
                    sensorDistance = content.trim();
                }
                break;
            case TAG_EXTENSION_ACCURACY_HORIZONTAL:
                if (content != null) {
                    accuracyHorizontal = content.trim();
                }
                break;
            case TAG_EXTENSION_ACCURACY_VERTICAL:
                if (content != null) {
                    accuracyVertical = content.trim();
                }
                break;
        }

        content = "";
    }

**Expected Solution**
![GpxTrackImporter complexity 33](https://user-images.githubusercontent.com/118127727/216791799-b51ebec3-41be-4c8c-97da-5c2d43df4dad.png)
